File: samples.vb/Northwind/EmployeeTerritory.vb
Imports System
Imports System.Data.Linq.Mapping

Namespace Northwind

   <Table(Name:="EmployeeTerritories")>
   Public Class EmployeeTerritory

      <Association(ThisKey:="EmployeeID", IsForeignKey:=True)>
      Public Property Employee As Employee

      <Column(IsPrimaryKey:=True)>
      Public Property EmployeeID As Integer

      <Association(ThisKey:="TerritoryID", IsForeignKey:=True)>
      Public Property Territory As Territory

      <Column(CanBeNull:=False, IsPrimaryKey:=True)>
      Public Property TerritoryID As String

   End Class

End Namespace

File: samples.cs/Northwind/EmployeeTerritory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Data.Linq.Mapping;

namespace Samples.CSharp.Northwind {

   [Table(Name = "EmployeeTerritories")]
   public class EmployeeTerritory {

      [Column(IsPrimaryKey = true)]
      public int EmployeeID { get; set; }

      [Column(CanBeNull = false, IsPrimaryKey = true)]
      public string TerritoryID { get; set; }

      [Association(ThisKey = "EmployeeID", IsForeignKey = true)]
      public Employee Employee { get; set; }

      [Association(ThisKey = "TerritoryID", IsForeignKey = true)]
      public Territory Territory { get; set; }
   }
}
